src/xcompact.f90

```diff
@@ -43,9 +43,13 @@ program xcompact
   integer, dimension(3) :: dims_global
   integer, dimension(3) :: nproc_dir = 0
   real(dp), dimension(3) :: L_global
+  character(3) :: poisson_solver_type
+  character(32) :: backend_name
   integer :: nrank, nproc, ierr
+  logical :: use_2decomp
 
   namelist /domain_params/ L_global, dims_global, nproc_dir, BC_x, BC_y, BC_z
+  namelist /solver_params/ poisson_solver_type
```

Comment:
What compiler was it with? it seems to be working fine with gnu. Could you double check before I move it to a new namelist?

Regardless of this, I think longer term it would be good to read the input file only once and make it populate an `option` or `parameter` object that will have everything. That way it can be accessed easily by the codebase without re-reading the file

Reply:
ah no, my bad, it also doesn't work with GNU, I will fix that.

Reply:
Just done this by creating a reader for the solver input. Seemed cleaner than alternatives. It will probably be good to do something similar for the domain parameters but it is a bit out of the scope of this PR.